observability/src/main/java/io/grpc/observability/ObservabilityConfig.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import io.grpc.internal.JsonParser;
+import io.grpc.internal.JsonUtil;
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/** gRPC Observability configuration processor. */
+final class ObservabilityConfig {
+  private static final String CONFIG_ENV_VAR_NAME = "GRPC_CONFIG_OBSERVABILITY";
+
+  private boolean enableCloudLogging = true;
+  private String destinationProjectId = null;
+  private LogFilter[] logFilters;
+  private GrpcLogRecord.EventType[] eventTypes;
+
+  public static class LogFilter {
+    public String pattern;
+    public Integer headerBytes;
+    public Integer messageBytes;
+  }
+
+  ObservabilityConfig() {
+  }
+
+  void read() throws IOException {
```

Comment:
I suggest you make this a static factory method that returns a new ObservabilityConfig instance. That way the client does not need to separately create a new instance and then call read(). Also eliminates the possibility of an instance from being in an intermediate/invalid state where it is created but has no data read into it.